internal/pkg/manifest/workload.go

```diff
@@ -66,6 +74,40 @@ type Image struct {
 	DependsOn    map[string]string `yaml:"depends_on,flow"` // Add any sidecar dependencies.
 }
 
+type imageTransformer struct{}
+
+// Transformer implements customized merge logic for Image field of manifest.
+// It merges `DockerLabels` and `DependsOn` in the default manager (i.e. with configurations mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+// And then overrides both `Build` and `Location` fields at the same time with the src values, given that they are non-empty themselves.
+func (t imageTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
```

Comment:
What's the reason why the `image` field override doesn't work today and we need a transformer?

Reply:
Currently, if the dst has only Build, and src has Location, the merged result will have both Build and Location non-empty, while Build and Location should be mutually exclusive.